blog-post/blog-post.md

```diff
@@ -0,0 +1,382 @@
+# Introducing Core Solidity
+
+Solidity earned its place as the de facto language for smart contracts by 
+prioritizing developer experience. Since its initial design, it favored 
+high-level abstractions which are well-known by programmers.
+While safety was always a goal, it was the history of contract exploits that 
+forged it over the years. This focus on security transformed Solidity, making 
+it stricter over time. These necessary changes, however, came with consequences. 
+The language now exhibits the scars of its accumulated complexity and a type system 
+that has become a barrier to its own evolution. Foundational features like 
+generics (aka parametric polymorphism) remain out of reach, as integrating them 
+into the current architecture is both difficult and dangerously prone to 
+introduce critical flaws.
+
+Aiming to break this impasse, the Argot collective decided to embark on a 
+fundamental rebuild, which is named Core Solidity: a project to define a new, 
+foundational version of the language designed for security, clarity, and 
+future growth from the ground up.
+
+
+## What is Core Solidity?
```

Comment:
Looks like the post no longer mentions SAIL. IMO we should say here that this is what we're describing. And later also way which bits are just syntax sugar or stdlib.